nixos/modules/services/mail/dovecot.nix

```diff
@@ -297,10 +298,15 @@ in
 
 
   config = mkIf cfg.enable {
-
     security.pam.services.dovecot2 = mkIf cfg.enablePAM {};
 
-    services.dovecot2.protocols =
+    security.dhparams = mkIf (! isNull(cfg.sslServerCert)) {
+      enable = true;
+      params = {
+        dovecot2 = 4096;
```

Comment:
I'm not sure we need to set 4096 bits for dhparams here by default, that's *very* slow to generate and there's not much value in having it so high, 2048 should be enough for most use cases, and 3072 for the paranoid (like me). See for instance https://github.com/certbot/certbot/pull/4973, where Let's Encrypt gives hardcoded 2048-bit DH params (which is way worse in security than what we have in NixOS with the DH params auto-generated, as they aren't used by others)